### DISCUSSION TITLE
Intercept dagsters termination signal and forward to execution engine

### DISCUSSION THREAD

**Discussion by geoHeil** [dagster-io/dagster#18459] [Q&A] [answered]
How can I forward the terminate signal of dagster and forward it to the execution engine?

When clicking the terminate button in the UI the databricks spark job launched with dagster-pipes via `jobs.SubmitTask.from_dict` will continue to run.

Where can I register a termination hook to relay the singal?

**Comment by geoHeil** [comment]
https://github.com/dagster-io/dagster/discussions/13986

links has some suggestions

```
from dagster import DagsterExecutionInterruptedError, op

@op
def my_op():
  try:
    # work goes here
  except DagsterExecutionInterruptedError:
    # perform any cleanup here

```

however - how is this applicable to the dagster-pipes launched databricks run?

https://docs.dagster.io/guides/dagster-pipes/databricks

```
@asset
def databricks_asset(
    context: AssetExecutionContext, pipes_databricks: PipesDatabricksClient
):
    task = jobs.SubmitTask.from_dict(
        {
            # workers.
            "new_cluster": {
                "spark_version": "12.2.x-scala2.12",
...
            "task_key": "some-key",
            "spark_python_task": {
                "python_file": "dbfs:/my_python_script.py",  # location of target code file
                "source": jobs.Source.WORKSPACE,
            },
        }
    )

    return pipes_databricks.run(
        task=task,
        context=context,
        extras=extras,
    ).get_materialize_result()
```

as `    task = jobs.SubmitTask.from_dict(` is not aware of the cluster ID that is only launched later in ```pipes_databricks.run(```

In order to send a termination request in the `DagsterExecutionInterruptedError` similar to: 

```
# Specify the run ID of the job/task you want to terminate
run_id = "<your-run-id>"

# Terminate the job
client.jobs.cancel_run(run_id)

```

**Reply by alangenfeld** [reply, marked as answer]
The way the pipes clients are setup currently we would need to handle this in the `run` method - filed a issue for that here https://github.com/dagster-io/dagster/issues/18570. 

In the interim you could do this in an implementation that does not use the existing client 
https://docs.dagster.io/guides/dagster-pipes/dagster-pipes-details-and-customization
